unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/Debugging/Performance/ProfilerRecordsService.cs

```diff
@@ -0,0 +1,105 @@
+
+        private ProfilerRecorder mainThreadTimeRecorder;
+
+        private ProfilerRecorder drawCallsRecorder;
+        private ProfilerRecorder reservedMemoryRecorder;
+        private ProfilerRecorder usedMemoryRecorder;
+        private ProfilerRecorder gcAllocatedInFrameRecorder;
+
+        private bool isRecordingAdditionalProfilers;
+
+        public float LastFrameTimeInSec => mainThreadTimeRecorder.LastValue * 1e-9f; // [sec]
+        public float LastFrameTimeInMS => mainThreadTimeRecorder.LastValue * 1e-6f; // [sec]
+        public float LastFPS => 1 / LastFrameTimeInSec;
+
+        public (float FrameTime, float FPS) AverageData
+        {
+            get
+            {
+                float frameTime = GetRecorderFrameAverage(mainThreadTimeRecorder) * 1e-6f; // [ms]
+                return (FrameTime: frameTime, FPS: 1000 / frameTime);
+            }
+        }
+
+        // Additional recordings
+        public long GcAllocatedInFrame => gcAllocatedInFrameRecorder.LastValue;
+        public long UsedMemory => usedMemoryRecorder.LastValue;
+        public long ReservedMemory => reservedMemoryRecorder.LastValue;
+        public long DrawCalls => drawCallsRecorder.LastValue;
+
+        public ProfilerRecordsService()
+        {
+            mainThreadTimeRecorder = ProfilerRecorder.StartNew(ProfilerCategory.Internal, "Main Thread", CAPACITY);
+            samples = new List<ProfilerRecorderSample>(CAPACITY);
+        }
+
+        public void Initialize() { }
+
+        public void Dispose()
+        {
+            mainThreadTimeRecorder.Dispose();
+
+            drawCallsRecorder.Dispose();
+            reservedMemoryRecorder.Dispose();
+            usedMemoryRecorder.Dispose();
+            gcAllocatedInFrameRecorder.Dispose();
+        }
+
+        public void RecordAdditionalProfilerMetrics()
+        {
+            if (isRecordingAdditionalProfilers) return;
+
+            drawCallsRecorder = ProfilerRecorder.StartNew(ProfilerCategory.Render, "Draw Calls Count");
+            reservedMemoryRecorder = ProfilerRecorder.StartNew(ProfilerCategory.Memory, "Total Reserved Memory");
+            usedMemoryRecorder = ProfilerRecorder.StartNew(ProfilerCategory.Memory, "Total Used Memory");
+            gcAllocatedInFrameRecorder = ProfilerRecorder.StartNew(ProfilerCategory.Memory, "GC Allocated In Frame");
+
+            isRecordingAdditionalProfilers = true;
+        }
+
+        public void StartRecordGCAllocatedInFrame()
+        {
+            if (isRecordingAdditionalProfilers)
+                return;
+
+            gcAllocatedInFrameRecorder = ProfilerRecorder.StartNew(ProfilerCategory.Memory, "GC Allocated In Frame");
+        }
+
+        public void StopRecordGCAllocatedInFrame()
+        {
+            if (isRecordingAdditionalProfilers)
+                return;
+
+            gcAllocatedInFrameRecorder.Dispose();
+        }
+
+        private float GetRecorderFrameAverage(ProfilerRecorder recorder)
```

Comment:
This one is only used in a specific scenario - when debug window is open (for example via `/showfps` command in chat). Currently the buffer set to 15 and needed only to prevent stuttering of the value in the view (we can increase it to 30). With 15 the overhead is minimal (taking into account it was recorded with DeepProfiler overhead)
![image](https://user-images.githubusercontent.com/35366872/224769531-91055998-3881-468e-944e-4568f30bfedd.png)

Reply:
`unsafe` version was more performant, so I changed to it (and also added comments for awarness)

Benchmark:
- for 200 samples with `List`
![image](https://user-images.githubusercontent.com/35366872/224775917-7a425065-8457-4837-a814-dc0795eb96ed.png)
- versus 200 samples on `unsafe`
![image](https://user-images.githubusercontent.com/35366872/224776010-867b16e6-8f93-4265-b6cb-f78998b37f4e.png)